app/RouteHandlers/AddDonationHandler.php

```diff
@@ -183,4 +186,33 @@ private function isSubmissionAllowed( Request $request ) {
 		return true;
 	}
 
+	private function collectPiwikTrackingEvents( array $sessionTrackingData, Donation $donation ) {
+		$piwikTracking = new PiwikCustomTracking();
+		if ( isset( $sessionTrackingData['paymentType'] ) ) {
+			$piwikTracking->setCustomVariable(
+				1,
```

Comment:
I also thought about what @gbirke wrote, but found it silly, too. The class now has constants for the variable IDs and a mapping array. It will throw an exception if `setCustomVariable` is called with an undefined ID.